avocado/core/varianter.py

```diff
@@ -467,6 +467,15 @@ def to_str(self, summary=0, variants=0, **kwargs):
         :param kwargs: Other free-form arguments
         :rtype: str
         """
+        if self._no_variants == 0:  # No variants, only defaults:
+            out = []
+            if summary:
+                out.append("No variants available, using defaults only")
+            if variants:
+                out.append("\n".join(variant_to_str(next(self.itertests()),
+                                                    variants - 1, kwargs,
+                                                    self.debug)))
+            return "\n\n".join(out)
```

Comment:
The problem here is that it will expose internal implementation to the user in case of no variants:
```
~/git/avocado (ldoktor-system-wide)$ > /tmp/test.yaml
~/git/avocado (ldoktor-system-wide)$ avocado variants -m /tmp/test.yaml
Variant None:    /
```
`None` should not be in a valid user-facing output.